### ISSUE TITLE
integrate(x/(a*x+b), x) gives wrong answer

### ISSUE THREAD

**Issue by rtrwalker** [sympy/sympy#10488]
Using sympy version 0.7.5, for the following code:

```
import sympy
x, a, b  = sympy.symbols('x,a,b')
print(sympy.integrate(x/(a*x+b), x))
```
I get:

`x/a - b*log(a**2*x + a*b)/a**2`

which is wrong.  I should get:
`x/a - b*log(a*x + b)/a**2`

I'm not sure where the extra 'a' multiple within the log comes from.

Same thing happens on http://live.sympy.org/ which is sympy version 0.7.6

**Comment by rtrwalker** [comment]
I've narrowed it down a bit.  Following code produces the same wrong result as above:
```
import sympy
from sympy.integrals.rationaltools import ratint
x, a, b  = sympy.symbols('x,a,b')
print(ratint(x/(a*x+b), x))
```

As far as I can see the `a**2*x+a*b` term comes in after the line `g, h = ratint_ratpart(p, q, x)` which is line# 52(sympy version 0.7.5) of `sympy.integrals.rationaltools`  in the ratint function.    After that I get lost in Poly functions.

**Comment by shubhamtibra** [comment]
Technically the answer is not wrong as `x/a - b*log(a**2*x + a*b)/a**2` and  `x/a - b*log(a*x + b)/a**2` differ by a constant which is `b*log(a)/a**2` . Both are correct in indefinite integration.
I think after partial fraction ,sympy is integrating `1/(a*(a*x+b))` as a single term which gives `x/a - b*log(a**2*x + a*b)/a**2` .

**Comment by rtrwalker** [comment]
@shubhamtibra you are correct.  If I do it by hand with the substitution t=a*x+b then my 'extra' constant is
b**2/a**2.  It is just annoying that the superfluous constant is left it.  If I put `x/(A x +B)` into http://www.wolframalpha.com/ the indefinite integral does not have the extra constant (just a "+constant" term).  It would be nice if sympy gave the same answer.

**Comment by shubhamtibra** [comment]
I will look into this.

**Comment by asmeurer** [comment]
Perhaps a simple solution would be a post-processor in ratint that removes constant terms from the logarithms (or it may be easier to do it as a pre-processor, I'm not sure). I agree that the solution is not wrong, but it's also nice for integrate to return simpler answers when it can, and there are definitely some known issues with both ratint and rischintegrate including redundant constants in logarithms.

**Comment by shubhamtibra** [comment]
@asmeurer Can you explain more on how to do that?

**Comment by Toshad** [comment]
Has this issue been resolved?? I would like to work on this.

**Comment by asmeurer** [comment]
It hasn't been fixed, but I don't know if @shubhamtibra is working on it.

**Comment by shubhamtibra** [comment]
I'm not working on it right now. You can take it @Toshad .

**Comment by Toshad** [comment]
Is there a function which will return a list of factors of an expression? Currently I am trying to make the list from the output of factor(), but I am not sure it will work in all cases.

**Comment by jksuom** [comment]
There is ``Poly.factor_list`` in ``sympy.polys.polytools``.
```
        >>> from sympy import Poly
        >>> from sympy.abc import x, y

        >>> f = 2*x**5 + 2*x**4*y + 4*x**3 + 4*x**2*y + 2*x + 2*y

        >>> Poly(f).factor_list()
        (2, [(Poly(x + y, x, y, domain='ZZ'), 1),
             (Poly(x**2 + 1, x, y, domain='ZZ'), 2)])
```

**Comment by Toshad** [comment]
Thanks.

**Comment by shubhamtibra** [comment]
@asmeurer Were you talking about traversing the final expression to find instances of `log` and use `expand_log` to seperate out the `log` terms?

**Comment by asmeurer** [comment]
It should be possible to do this at a much lower level in the algorithm, where it actually creates the logarithms.

**Comment by ashutoshsaboo** [comment]
Hi @asmeurer . I have been looking into this since a pretty long time. 

What I feel till now is, if we want to resolve for logarithms in general only, then we can find the log terms in the expression, and then use expand_log on them. But the thing with this is that, it doesn't solve our purpose for any general integral value.

The best I can think is, to return the simplified answer, by operation the simplify() function on the value of the integral that is returned, in case of the risch algorithm case. Because, in that case it simplifies the expressions, so easily, and also puts brackets, and hence it becomes easier for us, as users, to interpret what could have been the constant term that added up in case of the indefinite integral. 

I understand that it doesn't solve our issue, of returning the constant - 'c' term in case of indefinite integration, but I guess, this also, solves our problem, as the users can then easily distinguish and identify the answer easily, once the simplify() function puts up the necessary brackets, and factors it up.

In all other cases, wherein the integrand is a polynomial or so, simplify() will perfectly do the job. In case of log terms only, this is a simple trade-off that we need to take.

Any views on this? @asmeurer @jksuom 

Could someone help me build on this? Or suggest, in case of any alternate procedure, that can solve us this problem, and hence this issue as well?

Thanks! :)

**Comment by ashutoshsaboo** [comment]
@asmeurer @jksuom 

Hi. I looked into this issue more closely. 

With the addition of 2 simple conditions, it now gives the simplest possible answer for integral answers involving 'log' which is handled by the expand_log, and also other type of answers are simplified to the maximum possible by extent by the simplify() function. 

Please find the PR, and suggest in case if there are any other additions in the code required?

Thanks :smile: !

**Comment by jksuom** [comment]
Integration of rational functions is implemented in ``ratint``. [This line](https://github.com/sympy/sympy/blob/master/sympy/integrals/rationaltools.py#L95) (and a similar one below) will create the ``log`` terms. It is possible to remove extra factors such as ``a`` in this example by adding the following line above those lines.
``    _, h = h.primitive()``

**Comment by ashutoshsaboo** [comment]
@jksuom Thanks a lot! I'll open a new PR with the required changes, and the updated test files as well!

 :smile:

**Comment by ashutoshsaboo** [comment]
@asmeurer @jksuom 

I've opened up the new PR, that fixes this issue - #10585 . Please have a look into. Are there any other additions required in the code? Or can this be merged now?

Thanks! :smile:
